ReactiveCocoa/Swift/Property.swift

```diff
@@ -240,12 +240,18 @@ public func <~ <P: MutablePropertyType>(property: P, signal: Signal<P.Value, NoE
 public func <~ <P: MutablePropertyType>(property: P, producer: SignalProducer<P.Value, NoError>) -> Disposable {
 	let disposable = CompositeDisposable()
 
-	producer.startWithSignal { signal, signalDisposable in
-		property <~ signal
-		disposable += signalDisposable
-
-		disposable += property.producer.startWithCompleted {
+	disposable += property.producer.startWithCompleted {
+		disposable.dispose()
+	}
+	
+	disposable += producer.start { [weak property] event in
```

Comment:
I tend to use `weak` because it is much safer. With `unowned` we have to be absolutely sure closure is not executed after property is deallocated. I'm not ready to say may it happen or not.